app/controllers/sessions_controller.rb

```diff
@@ -8,11 +8,16 @@ def create
     if user.authenticate(params[:password])
       session[:user_id] = user.id
       flash[:success] = "Welcome #{user.user_detail.name}"
-      redirect_to '/'
+      dynamic_redirect
     else
       flash[:error] = "Invalid Credentials"
       render :new
     end
   end
 
-end
\ No newline at end of file
+  def dynamic_redirect
+    redirect_to "/profile" if current_user.role == "default"
+    redirect_to "/merchant/dashboard" if current_user.role == "merchant"
+    redirect_to "/admin/dashboard" if current_user.role == "admin"
```

Comment:
I like this. How about if we make the `dynamic_redirect` method `private`. Probably not a good idea to expose those redirects outside the controller.

Reply:
good idea, will make the private